### PR TITLE
Standard search operator

### PR DESCRIPTION
**Pull request by chrizzor**
This PR adds the possiblity to configure the standard search operator. (e.g. mainQueryOperator = AND)

### CODE REVIEW

File: Classes/Service/SolrServiceProvider.php [hunk 1]
```diff
@@ -627,6 +627,7 @@ protected function createQuery(): void
         $this->query = $this->connection->createSelect();
         $this->addFeatures();
         $this->addTypoScriptFilters();
+        $this->addMainQueryOperator();
```
Comment (ipf) [inline Classes/Service/SolrServiceProvider.php:630]: The naming of "MainQuery" is quite misleading here - I guess "DefaultQuery" is what expresses the method best

File: Classes/Service/SolrServiceProvider.php [hunk 2]
```diff
@@ -1147,4 +1148,16 @@ protected function tagForFacet(string $facetID): string
     {
         return 'facet-'.$facetID;
     }
+
+    /*
+     * Set configured main query operator. Defaults to 'AND'.
+     */
+    private function addMainQueryOperator()
+    {
+        $mainQueryOperator = 'AND';
```
Comment (ipf) [inline Classes/Service/SolrServiceProvider.php:1157]: This introduces a previously only implicit set operator `AND`. Why not leave this untouched and only do something in the `if`-clause below?

File: Classes/Service/SolrServiceProvider.php [hunk 3]
```diff
@@ -1147,4 +1148,16 @@ protected function tagForFacet(string $facetID): string
...
+        if (isset($this->settings['mainQueryOperator'])) {
+            $mainQueryOperator = $this->settings['mainQueryOperator'];
```
Comment (ipf) [inline Classes/Service/SolrServiceProvider.php:1159]: How should a developer know about this setting? It's not documented anywhere.
So, please put this in the documentation as well, and maybe provide examples in the TypoScript code.
Do we have to evaluate/trim the setting or silently fail if the operator is not supported by Solr (e.g `FOO` as setting)?

**Review by ipf — CHANGES_REQUESTED** [review]

**Comment by ipf** [comment]
Hi and thanks a lot for the PR.
I added a review.
Please provide an english commit message.

**Comment by mh-goe** [comment]
Seems good to me. Thx for the contribution, @chrizzor

**Pull request merged**